archetype/test/setup.js

```diff
@@ -3,8 +3,8 @@
 const TrailsApp = require('trails')
 
 before(() => {
-  global.app = new TrailsApp(require('../../'))
-  return global.app.start()
+  global.app = new TrailsApp(require('../'))
+  return global.app.start().catch(global.app.stop)
```

Comment:
The app will already try to `.stop()` itself if it encounters an error. I think this may cause more issues than it solves.

Can you try forcing an app error on load and see how this logic behaves?